crates/net/network/src/discovery.rs

```diff
@@ -40,7 +40,7 @@ pub struct Discovery {
     /// All nodes discovered via discovery protocol.
     ///
     /// These nodes can be ephemeral and are updated via the discovery protocol.
-    discovered_nodes: LruMap<PeerId, SocketAddr>,
+    discovered_nodes: LruMap<PeerId, (SocketAddr, SocketAddr)>,
```

Comment:
this should document the addresses.

I think we should consider a helper type in the peers crate like `PeerAddr {tcp, udp}`